pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -676,32 +677,82 @@ def _append_if_else_circuit(
     cargs: list[Clbit],
 ) -> None:
     # Get two pytket circuits which implement the true_body and false_body.
+    # Note that else_circ can be None if no false_body is specified.
     if_circ, else_circ = _pytket_circuits_from_ifelseop(
         if_else_op, outer_builder, qargs, cargs
     )
-    # else_circ can be None if no false_body is specified.
-    if isinstance(if_else_op.condition[0], Clbit):
-        if len(bits) != 1:
-            raise NotImplementedError("Conditions on multiple bits not supported")
+
+    if isinstance(if_else_op.condition, (Unary | Binary)):
+        raise NotImplementedError(
+            "Handling of Unary and Binary conditions is not supported yet."
+        )
+    if isinstance(if_else_op.condition, Var) and isinstance(
+        if_else_op.condition.var, Clbit
+    ):
+        condition_bits = [
+            bit
+            for bit in bits
+            if bit.reg_name == if_else_op.condition.var._register.name  # noqa: SLF001
+            and bit.index[0] == if_else_op.condition.var._index  # noqa: SLF001
+        ]
+
+        if len(condition_bits) == 0:
+            raise ValueError(
+                "Failed to find any pytket Bit matching Qiskit Clbit in condition for IfElseOp."
+            )
+
+        # In this case, if_else_op.condition is a single tuple of shape (Clbit, value)
+        outer_builder.tkc.add_circbox(
+            circbox=CircBox(if_circ),
+            args=if_circ.qubits + if_circ.bits,  # type: ignore
+            condition_bits=condition_bits,
+            condition_value=1,
+        )
+        # If we have an else_circ defined, add it to the circuit
+        if else_circ is not None:
+            outer_builder.tkc.add_circbox(
+                circbox=CircBox(else_circ),
+                args=else_circ.qubits + else_circ.bits,  # type: ignore
+                condition_bits=condition_bits,
+                condition_value=0,
```

Comment:
Similarly: this will be true iff all the condition bits are unset. I think the condition should be that any of them is unset?